tests/components/system_log/test_init.py

```diff
@@ -195,6 +217,17 @@ async def test_critical(hass, hass_ws_client):
     assert_log(log, "", "critical message", "CRITICAL")
 
 
+async def test_critical_with_missing_format_args(hass, hass_ws_client):
+    """Test that critical messages with missing format args are logged and retrieved correctly."""
+    await async_setup_component(hass, system_log.DOMAIN, BASIC_CONFIG)
+    await hass.async_block_till_done()
+
+    try:
+        _LOGGER.critical("critical message %s = %s", "one_but_needs_two")
+    except TypeError:
+        pass
```

Comment:
Where is the assertion?

Reply:
Critical and warning will both do the same, but there must be at least one valid argument, and at least one missing argument to trigger the exception in logging since the base logging module has a guard to check for `self.args`:

https://github.com/python/cpython/blob/ba6492e2a705ebbe140fdf8734114156e55b3c9a/Lib/logging/__init__.py#L367

This will hit this exception trap when not run under pytest (which patches logging)
https://github.com/home-assistant/core/pull/80645/files#diff-d7d18627b016b0d21ad36d10f06b43146746bf30e77d78ecf7851c579f6795f1R101

Reply:
Is the test not testing the code then?

Reply:
Its not hitting that line as its not possible to do so under pytest since the logging is trapped

Reply:
To disable the pytest capture or logging plugin we'd need a separate CI job, it seems. We could add that and just run the logger tests, if we think it's important.

Reply:
I think the added maintenance burden doesn't justify the cost of adding another CI job.

Additionally I think the risk of someone modifying this and causing a regression is low.

I'm going to add some additional comments and details in the doc strings to mitigate that risk.

Reply:
Sounds good.

Reply:
76d69671ef714bc38ff9859d3d2938e2e3c63724

Reply:
I still think we should remove this test, since it doesn't seem to do anything.

Reply:
https://github.com/home-assistant/core/pull/80739